daffodil-runtime1/src/main/scala/org/apache/daffodil/layers/Base64Transformer.scala

```diff
@@ -0,0 +1,43 @@
+package org.apache.daffodil.layers
+
+import org.apache.daffodil.schema.annotation.props.gen.LayerLengthKind
+import org.apache.daffodil.schema.annotation.props.gen.LayerLengthUnits
+import org.apache.daffodil.util.Maybe
+import org.apache.daffodil.processors.TermRuntimeData
+import org.apache.daffodil.processors.LayerLengthInBytesEv
+import org.apache.daffodil.processors.LayerTerminatorEv
+import org.apache.daffodil.processors.LayerCharsetEv
+
+class Base64MIMETransformer(layerCharsetEv: LayerCharsetEv, layerTerminatorEv: LayerTerminatorEv)
+  extends LayerTransformer() {
+
+  override def wrapLayerDecoder(jis: java.io.InputStream) = {
+    val b64 = java.util.Base64.getMimeDecoder().wrap(jis)
+    b64
+  }
+
+}
+
+object Base64MIMETransformerFactory
```

Comment:
Add scaladoc: newInstance method digests the various layering related arguments to decide how to configure and which layer transformer to create, then constructs it appropriately.